`InvalidDefinitionException` when calling `mapper.createObjectNode().putPOJO`

Hello,

When running the code below I get the exception:
```
Exception in thread "main" java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.LocalDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
	at com.fasterxml.jackson.databind.node.InternalNodeMapper.nodeToPrettyString(InternalNodeMapper.java:40)
	at com.fasterxml.jackson.databind.node.BaseJsonNode.toPrettyString(BaseJsonNode.java:141)
	at Main.main(Main.java:16)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.LocalDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.LocalDateTime` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling

	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1300)
	at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
	at com.fasterxml.jackson.databind.SerializerProvider.defaultSerializeValue(SerializerProvider.java:1142)
	at com.fasterxml.jackson.databind.node.POJONode.serialize(POJONode.java:115)
	at com.fasterxml.jackson.databind.node.ObjectNode.serialize(ObjectNode.java:328)
	at com.fasterxml.jackson.databind.ser.std.SerializableSerializer.serialize(SerializableSerializer.java:39)
	at com.fasterxml.jackson.databind.ser.std.SerializableSerializer.serialize(SerializableSerializer.java:20)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1518)
	at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1219)
	at com.fasterxml.jackson.databind.ObjectWriter.writeValueAsString(ObjectWriter.java:1086)
	at com.fasterxml.jackson.databind.node.InternalNodeMapper.nodeToPrettyString(InternalNodeMapper.java:38)
	... 2 more
```
2.11.4 works as expected.
2.13.0-rc2 fails with the same error.

Main.java
```java
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;

import java.time.LocalDateTime;

public class Main {

    public static void main(String[] args) {
        ObjectMapper mapper = JsonMapper.builder()
            .addModule(new JavaTimeModule())
            .build();

        ObjectNode node = mapper.createObjectNode().putPOJO("test", LocalDateTime.now());
        System.out.println(node.toPrettyString());
    }

}
```

build.gradle
```groovy
plugins {
    id 'java'
}

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
    implementation platform("com.fasterxml.jackson:jackson-bom:2.12.5")
    implementation "com.fasterxml.jackson.core:jackson-databind"
    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310"
}
```

I've tried using AdoptOpenJDK (OpenJ9) 11.0.11 and AdoptOpenJDK (Hotspot) 14.0.2 and I get the same result on both.

--- a/src/main/java/com/fasterxml/jackson/databind/node/InternalNodeMapper.java
+++ b/src/main/java/com/fasterxml/jackson/databind/node/InternalNodeMapper.java
@@ -134,6 +134,14 @@ final class InternalNodeMapper {
                         stack.push(currIt);
                         currIt = value.elements();
                         g.writeStartArray(value, value.size());
+                    } else if (value instanceof POJONode) {
+                        // [databind#3262] Problematic case, try to handle
+                        try {
+                            value.serialize(g, _context);
+                        } catch (IOException | RuntimeException e) {
+                            g.writeString(String.format("[ERROR: (%s) %s]",
+                                    e.getClass().getName(), e.getMessage()));
+                        }
                     } else {
                         value.serialize(g, _context);
                     }

--- a/src/test/java/com/fasterxml/jackson/failing/POJONode3262Test.java
+++ b/src/test/java/com/fasterxml/jackson/failing/POJONode3262Test.java
@@ -4,11 +4,11 @@ import java.time.LocalDateTime;
 
 import com.fasterxml.jackson.databind.*;
 
-// [databind#3262]: not sure what could be done here. The issue is that
+// [databind#3262]: The issue is that
 // `JsonNode.toString()` will use internal "default" ObjectMapper which
 // does not (and cannot) have modules for external datatypes, such as
-// Java 8 Date/Time types. One possibility would be catch IOException for
-// POJONode, produce something like "ERROR: <ExceptionClass>" TextNode for that case?
+// Java 8 Date/Time types. So we'll catch IOException/RuntimeException for
+// POJONode, produce something like "[ERROR: (type) [msg]" TextNode for that case?
 public class POJONode3262Test extends BaseMapTest
 {
     private final ObjectMapper MAPPER = newJsonMapper();
@@ -17,7 +17,13 @@ public class POJONode3262Test extends BaseMapTest
     {
         JsonNode node = MAPPER.createObjectNode().putPOJO("test", LocalDateTime.now());
         String json = node.toString();
-
         assertNotNull(json);
+
+        JsonNode result = MAPPER.readTree(json);
+        String msg = result.path("test").asText();
+        assertTrue("Wrong fail message: "+msg,
+                msg.startsWith("[ERROR:"));
+        assertTrue("Wrong fail message: "+msg,
+                msg.contains("InvalidDefinitionException"));
     }
 }

